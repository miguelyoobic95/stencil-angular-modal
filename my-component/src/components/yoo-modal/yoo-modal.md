---
name: Modal
category: Core Components
---

Modals can be customized with multiple properties. Properties conform to the ModalEntry interface below:

```ts
export interface ModalEntry {
    title?: string;
    titleIcon?: string;
    hasHeader?: boolean;
    hasFooter?: boolean;
    footerText?: string;
    content?: HTMLElement;
    primaryButtonText?: string;
    secondaryButtonText?: string;
    cssClass?: string;
}
```

## Attributes
|Name|Type|Default|Description|
|---|---|---|---|
|`title`|string|   |the title of the modal, will only be displayed if `hasHeader` property is `true`|
|`titleIcon`|string| |the icon to display to the left of the title|
|`content`|HTMLElement| |the content inside the body of the modal|
|`hasHeader`|boolean|`true`|defines if the modal has a header|
|`hasFooter`|boolean|`false`|defines if the modal has a footer|
|`primaryButtonText`|string|`Confirm`  |the text inside the primary button, will only be displayed if `hasFooter` property is `true`|
|`secondaryButtonText`|string|`Cancel`|the text inside the secondary button, will only be displayed if `hasFooter` property is `true`|
|`cssClass`|string| |space separated classes to add to the modal|

### Basic

```yoo-modal-basic.html
    <yoo-modal title="Basic Modal">
        <div slot="modal-content">Test content</div>        
    </yoo-modal>
```

### With Footer

```yoo-modal-footer.html
    <yoo-modal class="accent" icon="yo-fire" title="Modal with footer" has-footer="true" primary-button-text="Confirm" secondary-button-text="Cancel">
        <div slot="modal-content">Test content</div>
    </yoo-modal>
```

### Without Header

```yoo-modal-no-header.html
    <yoo-modal has-footer="true" has-header="false">
        <div slot="modal-content">Test content</div>
    </yoo-modal>
```
## CSS
|Type|Name|Description|
|---|---|---|
|Accent|`accent`|yoo-modal accent|
|Stable|`stable`|yoo-modal stable|
|Success|`success`|yoo-modal success|
|Danger|`danger`|yoo-modal danger|
|Info|`info`|yoo-modal info|
|Warning|`warning`|yoo-modal warning|
|Gradient Accent|`gradient-accent`|yoo-modal gradient-accent|
|Gradient Stable|`gradient-grey`|yoo-modal gradient-grey|
|Gradient Success|`gradient-success`|yoo-modal gradient-success|
|Gradient Danger|`gradient-danger`|yoo-modal gradient-danger|
|Gradient Info|`gradient-info`|yoo-modal gradient-info|
|Gradient Warning|`gradient-warning`|yoo-modal gradient-warning|
|Fullscreen|`fullscreen`|yoo-modal fullscreen|
|Drawer|`drawer`|yoo-modal drawer|

```yoo-modal-danger.html
    <yoo-modal class="danger" icon="yo-fire" title="Modal danger" has-footer="true" primary-button-text="Confirm" secondary-button-text="Cancel">
        <div slot="modal-content">Test content</div>
    </yoo-modal>
```

```yoo-modal-success.html
    <yoo-modal class="success" icon="yo-fire" title="Modal success" has-footer="true" primary-button-text="Confirm" secondary-button-text="Cancel">
        <div slot="modal-content">Test content</div>
    </yoo-modal>
```

```yoo-modal-info.html
    <yoo-modal class="info" icon="yo-fire" title="Modal info" has-footer="true" primary-button-text="Confirm" secondary-button-text="Cancel">
        <div slot="modal-content">Test content</div>
    </yoo-modal>
```

```yoo-modal-warning.html
    <yoo-modal class="warning" icon="yo-fire" title="Modal warning" has-footer="true" primary-button-text="Confirm" secondary-button-text="Cancel">
        <div slot="modal-content">Test content</div>
    </yoo-modal>
```

```yoo-modal-stable.html
    <yoo-modal class="stable" icon="yo-fire" title="Modal stable" has-footer="true" primary-button-text="Confirm" secondary-button-text="Dismiss">
        <div slot="modal-content">Test content</div>
    </yoo-modal>
```

## Events
|Attr|Description|
|---|---|
|`onPrimaryButtonClicked`|called when the primary button is clicked|
|`onClose`|called when the modal is closed|