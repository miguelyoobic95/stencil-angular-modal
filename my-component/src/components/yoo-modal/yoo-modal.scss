@import "../../styles/_variables.scss";

@mixin color-modal($color, $color-light) {
    .outer-container {
        color: $text-color;
        border: none;
        .modal-header {
            background-color: $color-light;
            color: $color;
            border-bottom-color: $color;
        }
    }
}

@mixin scale-modal() {
    .outer-container {
        border-radius: 0rem;
        height: 100%;
        width: 100%;
        .modal-header {
            border-radius: 0rem;
            width: 100%;
            position: absolute;
        }
        .modal-body {
            padding-top: 5.0625rem;
        }
        .modal-footer {
            width: 100%;
            position: absolute;
            bottom: 0rem;
        }
    }
}

:host {
    position: absolute;
    z-index: 2000;
    top: 25%;
    right: 25%;
    -webkit-animation: scaleUp .5s cubic-bezier(0.165, 0.840, 0.440, 1.000) forwards;            
    -moz-animation: scaleUp .5s cubic-bezier(0.165, 0.840, 0.440, 1.000) forwards;
    -o-animation: scaleUp .5s cubic-bezier(0.165, 0.840, 0.440, 1.000) forwards;
    animation: scaleUp .5s cubic-bezier(0.165, 0.840, 0.440, 1.000) forwards;
    .outer-container {
        border: 1px solid cssvar(grey-10);
        background: cssvar(light);
        width: 37.5rem;
        border-radius: 0.3125rem;
        .modal-header {
            padding: 1.5625rem;
            height: 4.0625rem;
            @include border-top-radius(0.3125rem);
            border-bottom: 1px solid cssvar(grey-20);
            .modal-title{
                font-size: 1rem;
                font-weight: bold;
            }
            .icon {
                padding: 0.5rem;
            }
            .close-icon {
                cursor: pointer;
            }
        }
        .modal-body {
            padding: 0.5rem 1.5625rem;
            min-height: 10.9375rem;
        }
        .modal-footer {
            padding: 1.5625rem;
            height: 4.0625rem;
            border-top: 1px solid cssvar(grey-20);
            align-items: center;
            .footer-buttons {
                flex: 1 1 33%;
                .primary-button,
                .secondary-button {
                    padding: 0.5rem;
                }
            }
        }
    }
}

:host(.out) {
    -webkit-animation: scaleDown .5s cubic-bezier(0.165, 0.840, 0.440, 1.000) forwards;            
    -moz-animation: scaleDown .5s cubic-bezier(0.165, 0.840, 0.440, 1.000) forwards;
    -o-animation: scaleDown .5s cubic-bezier(0.165, 0.840, 0.440, 1.000) forwards;
    animation: scaleDown .5s cubic-bezier(0.165, 0.840, 0.440, 1.000) forwards;
}

@keyframes scaleUp {
    0% {
      transform:scale(.8) translateY(1000px);
      opacity:0;
    }
    100% {
      transform:scale(1) translateY(0px);
      opacity:1;
    }
}

@keyframes scaleDown {
    0% {
      transform:scale(1) translateY(0px);
      opacity:1;
    }
    100% {
      transform:scale(.8) translateY(1000px);
      opacity:0;
    }
}

@keyframes scaleFromRight {
    0% {
        transform:scale(.8) translateX(1000px);
        opacity:0;
      }
      100% {
        transform:scale(1) translateX(0px);
        opacity:1;
      }
}


:host(.drawer) {
    top: 0rem;
    right: 0rem;
    height: 100%;
    width: 33.33%;
    @include scale-modal();
    -webkit-animation: scaleFromRight .5s cubic-bezier(0.165, 0.840, 0.440, 1.000) forwards;            
    -moz-animation: scaleFromRight .5s cubic-bezier(0.165, 0.840, 0.440, 1.000) forwards;
    -o-animation: scaleFromRight .5s cubic-bezier(0.165, 0.840, 0.440, 1.000) forwards;
    animation: scaleFromRight .5s cubic-bezier(0.165, 0.840, 0.440, 1.000) forwards;
    .outer-container {
        .modal-header {
            flex-direction: row-reverse;
        }
    }
}

:host(.fullscreen) {
    top: 0rem;
    right: 0rem;
    height: 100%;
    width: 100%;
    @include scale-modal();
}

:host(.accent) {
@include color-modal(cssvar(accent), cssvar(accent-20));
}

:host(.danger) {
@include color-modal(cssvar(danger), cssvar(danger-20));
}

:host(.success) {
@include color-modal(cssvar(success-110), cssvar(success-20));
}

:host(.info) {
@include color-modal(cssvar(info), cssvar(info-20));
}

:host(.warning) {
@include color-modal(cssvar(warning), cssvar(warning-20));
}

:host(.stable) {
@include color-modal(cssvar(grey-100), cssvar(grey-20));
}