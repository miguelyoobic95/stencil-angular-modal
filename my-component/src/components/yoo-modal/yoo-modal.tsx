import { Component, Method, Element, Prop, EventEmitter, Event } from '@stencil/core';

@Component({
    tag: 'yoo-modal',
    styleUrl: 'yoo-modal.scss',
    scoped: true
})
export class YooModalComponent {

    @Element() el: HTMLElement;

    @Prop() title: string;
    @Prop() titleIcon: string;    
    @Prop() content: HTMLElement;
    @Prop() primaryButtonText: string = 'Confirm';
    @Prop() secondaryButtonText: string = 'Cancel';
    @Prop() hasHeader: boolean = true;
    @Prop() hasFooter: boolean = false;
    @Prop() footerText: string;
    @Prop() cssClass: string;

    @Event() onPrimaryButtonClicked: EventEmitter<boolean>;    
    @Event() onClose: EventEmitter<boolean>;

    handlePrimaryClick(event) {
        if (event) {
            this.onPrimaryButtonClicked.emit(true);
        }
    }

    @Method()
    close() {
        this.el.classList.add('out');
        // timeout is .1s lower than animation time
        setTimeout(() => { 
            this.onClose.emit(true);
            this.el.classList.remove('out');
        }, 400);
    }

    componentDidLoad() {
        if (this.content) {
            this.el.querySelector('div.modal-body').appendChild(this.content);
        }
    }
    
    componentDidUpdate() {
        if (this.content) {
            this.el.querySelector('div.modal-body').appendChild(this.content);
        }
    }

    render() {
        return ( 
            <div class="outer-container">
                {this.hasHeader ?
                    <div class="modal-header" attr-layout="row" attr-layout-align="space-between">
                        <div class="inner-header">
                            {this.titleIcon ? <span class="icon"><i class={this.titleIcon}></i></span> : null }
                            <span class="modal-title">{this.title}</span> 
                        </div>
                        <span class="close-icon" onClick={() => this.close()}><i class="yo-close"></i></span> 
                    </div> 
                : null }
                <div class="modal-body" attr-layout= "row">
                    <slot name="modal-content"/>
                </div>
                {this.hasFooter ? 
                    <div class="modal-footer" attr-layout="row">
                        {this.footerText ? <div class="footer-text"><span>{this.footerText}</span></div> : null}
                        <div class="footer-buttons" attr-layout="row" attr-layout-align="end center">
                            <div class="secondary-button"><yoo-button class="stable" onClick={() => this.close()} title={this.secondaryButtonText}></yoo-button></div>
                            <div class="primary-button"><yoo-button class="accent" onClick={(event: UIEvent) => this.handlePrimaryClick(event)} title={this.primaryButtonText}></yoo-button></div>
                        </div>
                    </div> 
                : null}
            </div>
        );
    }
}

export interface ModalEntry {
    title?: string;
    titleIcon?: string;
    hasHeader?: boolean;
    hasFooter?: boolean;
    footerText?: string;
    content?: HTMLElement;
    primaryButtonText?: string;
    secondaryButtonText?: string;
    cssClass?: string;
}

export function setModalProps(modal: any, props: ModalEntry) {
    modal.title = props.title;
    modal.titleIcon = props.titleIcon;    
    modal.hasFooter = props.hasFooter;
    modal.footerText = props.footerText;
    modal.hasHeader = props.hasHeader;
    modal.primaryButtonText = props.primaryButtonText;
    modal.secondaryButtonText = props.secondaryButtonText;
    modal.className += props.cssClass;
    modal.content = props.content;
    return modal;
}