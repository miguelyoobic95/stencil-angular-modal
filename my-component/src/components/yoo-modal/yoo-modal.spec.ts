import { render, flush } from '@stencil/core/testing';
import { YooModalComponent, setModalProps } from './yoo-modal';

describe('YooModalComponent', () => {
    it('should build', () => {
        expect(new YooModalComponent()).toBeTruthy();
    });

    describe('Snapshots', () => {
        it ('Should have a title and a header', async () => {
            let element = await render ({
                components: [YooModalComponent],
                html: '<yoo-modal title="this is text"></yoo-modal>'
            });
            expect(element).toMatchSnapshot();
        });

        it ('Should have no header', async () => {
            let element = await render ({
                components: [YooModalComponent],
                html: '<yoo-modal has-header="false"></yoo-modal>'
            });
            expect(element).toMatchSnapshot();
        });

        it ('Should have a title icon', async () => {
            let element = await render ({
                components: [YooModalComponent],
                html: '<yoo-modal title="this is text" title-icon="yo-fire"></yoo-modal>'
            });
            expect(element).toMatchSnapshot();
        });

        it ('Should have a footer', async () => {
            let element = await render ({
                components: [YooModalComponent],
                html: '<yoo-modal has-footer="true"></yoo-modal>'
            });
            expect(element).toMatchSnapshot();
        });

        it ('Should have content', async () => {
            let element = await render ({
                components: [YooModalComponent],
                html: `<yoo-modal>
                        <div slot="modal-content">Test content</div>
                        </yoo-modal>`
            });
            expect(element).toMatchSnapshot();
        });

        it ('Should have custom primary button text', async () => {
            let element = await render ({
                components: [YooModalComponent],
                html: `<yoo-modal primary-button-text="Hello"></yoo-modal>`
            });
            expect(element).toMatchSnapshot();
        });

        it ('Should have custom secondary button text', async () => {
            let element = await render ({
                components: [YooModalComponent],
                html: `<yoo-modal secondary-button-text="Hello"></yoo-modal>`
            });
            expect(element).toMatchSnapshot();
        });

        it ('Should be a full screen modal', async () => {
            let element = await render ({
                components: [YooModalComponent],
                html: `<yoo-modal css-class="fullscreen"></yoo-modal>`
            });
            expect(element).toMatchSnapshot();
        });

        it ('Should be a drawer modal', async () => {
            let element = await render ({
                components: [YooModalComponent],
                html: `<yoo-modal css-class="drawer"></yoo-modal>`
            });
            expect(element).toMatchSnapshot();
        });

        it('should close modal', async () => {
            jest.useFakeTimers();
            let element = await render ({
                components: [YooModalComponent],
                html: `<div><yoo-modal></yoo-modal></div>`
            });
            expect(element.querySelector('.out')).toBeNull();
            element.querySelector('yoo-modal').close();
            expect(element.querySelector('.out')).toBeTruthy();
            jest.runAllTimers();
            expect(element.querySelector('.out')).toBeNull();
            //ToDo yoo-modal should not exist? 
            //expect(element.querySelector('yoo-modal')).toBeNull();
        });

    });


    describe('rendering', () => {
        it ('Should render no title if hasHeader equals false', async () => {
            let element = await render ({
                components: [YooModalComponent],
                html: '<yoo-modal has-header="false" title="I should not exist"></yoo-modal>'
            });
            expect(element.getElementsByClassName('modal-header')).toHaveLength(0);
            expect(element.getElementsByClassName('modal-title')).toHaveLength(0);
        });

        it ('Should render no footer by default', async () => {
            let element = await render ({
                components: [YooModalComponent],
                html: '<yoo-modal></yoo-modal>'
            });
            expect(element.getElementsByClassName('modal-footer')).toHaveLength(0);
        });

        // TO DO: fix modal entry tests
        describe('modal-entry-rendering', () => {
            it ('Should render content', async () => {
                let element = await render ({
                    components: [YooModalComponent],
                    html: '<yoo-modal></yoo-modal>'
                });
                let htmlElement = document.createElement('div');
                htmlElement.innerHTML = `<div class="test">This is the test text</div>`;
                let modalProps = {content: htmlElement};
                element = await setModalProps(element, modalProps);
                flush(element);
                //expect(element.querySelector('div.test')).toBeTruthy();
                //expect(element.querySelector('div.test').innerHTML).toEqual('This is the test text');
            });

            it ('Should render title', async () => {
                let element = await render ({
                    components: [YooModalComponent],
                    html: '<yoo-modal></yoo-modal>'
                });
                let modalProps = {title: 'Arigato'};
                setModalProps(element, modalProps);
                expect(element.title).toEqual('Arigato');
                flush(element);
                // expect(element.getElementsByClassName('modal-title')).toHaveLength(1);
                // expect(element.getElementsByClassName('modal-title')[0].innerHTML).toEqual('Arigato');
            });
        });
        
    });
});