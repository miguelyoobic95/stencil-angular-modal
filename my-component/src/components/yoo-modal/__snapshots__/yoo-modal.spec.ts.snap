// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooModalComponent Snapshots Should be a drawer modal 1`] = `
<yoo-modal
  class="hydrated"
  css-class="drawer"
  data-yoo-modal-host=""
>
  <!---->
  <div
    class="outer-container"
    data-yoo-modal=""
  >
    <div
      attr-layout="row"
      attr-layout-align="space-between"
      class="modal-header"
      data-yoo-modal=""
    >
      <div
        class="inner-header"
        data-yoo-modal=""
      >
        
        <span
          class="modal-title"
          data-yoo-modal=""
        >
          
        </span>
      </div>
      <span
        class="close-icon"
        data-yoo-modal=""
      >
        <i
          class="yo-close"
          data-yoo-modal=""
        />
      </span>
    </div>
    <div
      attr-layout="row"
      class="modal-body"
      data-yoo-modal=""
      data-yoo-modal-slot=""
    />
    
  </div>
</yoo-modal>
`;

exports[`YooModalComponent Snapshots Should be a full screen modal 1`] = `
<yoo-modal
  class="hydrated"
  css-class="fullscreen"
  data-yoo-modal-host=""
>
  <!---->
  <div
    class="outer-container"
    data-yoo-modal=""
  >
    <div
      attr-layout="row"
      attr-layout-align="space-between"
      class="modal-header"
      data-yoo-modal=""
    >
      <div
        class="inner-header"
        data-yoo-modal=""
      >
        
        <span
          class="modal-title"
          data-yoo-modal=""
        >
          
        </span>
      </div>
      <span
        class="close-icon"
        data-yoo-modal=""
      >
        <i
          class="yo-close"
          data-yoo-modal=""
        />
      </span>
    </div>
    <div
      attr-layout="row"
      class="modal-body"
      data-yoo-modal=""
      data-yoo-modal-slot=""
    />
    
  </div>
</yoo-modal>
`;

exports[`YooModalComponent Snapshots Should have a footer 1`] = `
<yoo-modal
  class="hydrated"
  data-yoo-modal-host=""
  has-footer="true"
>
  <!---->
  <div
    class="outer-container"
    data-yoo-modal=""
  >
    <div
      attr-layout="row"
      attr-layout-align="space-between"
      class="modal-header"
      data-yoo-modal=""
    >
      <div
        class="inner-header"
        data-yoo-modal=""
      >
        
        <span
          class="modal-title"
          data-yoo-modal=""
        >
          
        </span>
      </div>
      <span
        class="close-icon"
        data-yoo-modal=""
      >
        <i
          class="yo-close"
          data-yoo-modal=""
        />
      </span>
    </div>
    <div
      attr-layout="row"
      class="modal-body"
      data-yoo-modal=""
      data-yoo-modal-slot=""
    />
    <div
      attr-layout="row"
      class="modal-footer"
      data-yoo-modal=""
    >
      
      <div
        attr-layout="row"
        attr-layout-align="end center"
        class="footer-buttons"
        data-yoo-modal=""
      >
        <div
          class="secondary-button"
          data-yoo-modal=""
        >
          <yoo-button
            class="stable"
            data-yoo-modal=""
            title="Cancel"
          />
        </div>
        <div
          class="primary-button"
          data-yoo-modal=""
        >
          <yoo-button
            class="accent"
            data-yoo-modal=""
            title="Confirm"
          />
        </div>
      </div>
    </div>
  </div>
</yoo-modal>
`;

exports[`YooModalComponent Snapshots Should have a title and a header 1`] = `
<yoo-modal
  class="hydrated"
  data-yoo-modal-host=""
  title="this is text"
>
  <!---->
  <div
    class="outer-container"
    data-yoo-modal=""
  >
    <div
      attr-layout="row"
      attr-layout-align="space-between"
      class="modal-header"
      data-yoo-modal=""
    >
      <div
        class="inner-header"
        data-yoo-modal=""
      >
        
        <span
          class="modal-title"
          data-yoo-modal=""
        >
          this is text
        </span>
      </div>
      <span
        class="close-icon"
        data-yoo-modal=""
      >
        <i
          class="yo-close"
          data-yoo-modal=""
        />
      </span>
    </div>
    <div
      attr-layout="row"
      class="modal-body"
      data-yoo-modal=""
      data-yoo-modal-slot=""
    />
    
  </div>
</yoo-modal>
`;

exports[`YooModalComponent Snapshots Should have a title icon 1`] = `
<yoo-modal
  class="hydrated"
  data-yoo-modal-host=""
  title="this is text"
  title-icon="yo-fire"
>
  <!---->
  <div
    class="outer-container"
    data-yoo-modal=""
  >
    <div
      attr-layout="row"
      attr-layout-align="space-between"
      class="modal-header"
      data-yoo-modal=""
    >
      <div
        class="inner-header"
        data-yoo-modal=""
      >
        <span
          class="icon"
          data-yoo-modal=""
        >
          <i
            class="yo-fire"
            data-yoo-modal=""
          />
        </span>
        <span
          class="modal-title"
          data-yoo-modal=""
        >
          this is text
        </span>
      </div>
      <span
        class="close-icon"
        data-yoo-modal=""
      >
        <i
          class="yo-close"
          data-yoo-modal=""
        />
      </span>
    </div>
    <div
      attr-layout="row"
      class="modal-body"
      data-yoo-modal=""
      data-yoo-modal-slot=""
    />
    
  </div>
</yoo-modal>
`;

exports[`YooModalComponent Snapshots Should have content 1`] = `
<yoo-modal
  class="hydrated"
  data-yoo-modal-host=""
>
  <!---->
  
                        
  
                        
  <div
    class="outer-container"
    data-yoo-modal=""
  >
    <div
      attr-layout="row"
      attr-layout-align="space-between"
      class="modal-header"
      data-yoo-modal=""
    >
      <div
        class="inner-header"
        data-yoo-modal=""
      >
        
        <span
          class="modal-title"
          data-yoo-modal=""
        >
          
        </span>
      </div>
      <span
        class="close-icon"
        data-yoo-modal=""
      >
        <i
          class="yo-close"
          data-yoo-modal=""
        />
      </span>
    </div>
    <div
      attr-layout="row"
      class="modal-body"
      data-yoo-modal=""
      data-yoo-modal-slot=""
    >
      <div
        slot="modal-content"
      >
        Test content
      </div>
    </div>
    
  </div>
</yoo-modal>
`;

exports[`YooModalComponent Snapshots Should have custom primary button text 1`] = `
<yoo-modal
  class="hydrated"
  data-yoo-modal-host=""
  primary-button-text="Hello"
>
  <!---->
  <div
    class="outer-container"
    data-yoo-modal=""
  >
    <div
      attr-layout="row"
      attr-layout-align="space-between"
      class="modal-header"
      data-yoo-modal=""
    >
      <div
        class="inner-header"
        data-yoo-modal=""
      >
        
        <span
          class="modal-title"
          data-yoo-modal=""
        >
          
        </span>
      </div>
      <span
        class="close-icon"
        data-yoo-modal=""
      >
        <i
          class="yo-close"
          data-yoo-modal=""
        />
      </span>
    </div>
    <div
      attr-layout="row"
      class="modal-body"
      data-yoo-modal=""
      data-yoo-modal-slot=""
    />
    
  </div>
</yoo-modal>
`;

exports[`YooModalComponent Snapshots Should have custom secondary button text 1`] = `
<yoo-modal
  class="hydrated"
  data-yoo-modal-host=""
  secondary-button-text="Hello"
>
  <!---->
  <div
    class="outer-container"
    data-yoo-modal=""
  >
    <div
      attr-layout="row"
      attr-layout-align="space-between"
      class="modal-header"
      data-yoo-modal=""
    >
      <div
        class="inner-header"
        data-yoo-modal=""
      >
        
        <span
          class="modal-title"
          data-yoo-modal=""
        >
          
        </span>
      </div>
      <span
        class="close-icon"
        data-yoo-modal=""
      >
        <i
          class="yo-close"
          data-yoo-modal=""
        />
      </span>
    </div>
    <div
      attr-layout="row"
      class="modal-body"
      data-yoo-modal=""
      data-yoo-modal-slot=""
    />
    
  </div>
</yoo-modal>
`;

exports[`YooModalComponent Snapshots Should have no header 1`] = `
<yoo-modal
  class="hydrated"
  data-yoo-modal-host=""
  has-header="false"
>
  <!---->
  <div
    class="outer-container"
    data-yoo-modal=""
  >
    
    <div
      attr-layout="row"
      class="modal-body"
      data-yoo-modal=""
      data-yoo-modal-slot=""
    />
    
  </div>
</yoo-modal>
`;
